### PR TITLE
Fix deepspeed examples bugs

### PR DESCRIPTION
**Pull request by cassieesvelt**
# Description
- Add timeout to deepspeed examples so runs will fail when running into an error.
- Fix hostfile error when process_count_per_gpu > 1
- change dockerfiles to use current acpt environments but pull from latest deepspeed.
# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: cli/jobs/deepspeed/deepspeed-autotuning/docker-context/Dockerfile [hunk 1]
```diff
@@ -1,39 +1,3 @@
...
-ENV SVDIR=/var/runit
-ENV WORKER_TIMEOUT=400
-EXPOSE 5001 8883 8888
-
-# support Deepspeed launcher requirement of passwordless ssh login
-RUN apt-get update
-RUN apt-get install -y openssh-server openssh-client
+FROM mcr.microsoft.com/azureml/curated/acpt-pytorch-1.11-py38-cuda11.3-gpu:9
+# Need latest deepspeed image
+RUN pip install git+https://github.com/microsoft/DeepSpeed.git@master
```
Comment (rtanase) [inline cli/jobs/deepspeed/deepspeed-autotuning/docker-context/Dockerfile:3]: why not install from pypi the latest version 0.8 ?

File: cli/jobs/deepspeed/deepspeed-autotuning/generate-yml.sh [hunk 2]
```diff
@@ -2,13 +2,15 @@
 # Generate key
 ssh-keygen -t rsa -f './src/generated-key' -N ''
 
-# Generate yaml file with key path
+# Pre-set process_count_per_instance so it can be passed into deepspeed via bash script.
+process_count_per_instance=2
```
Comment (rtanase) [inline cli/jobs/deepspeed/deepspeed-autotuning/generate-yml.sh:6]: not necessarily a big deal, but should this be renamed to number of GPUs per node?

File: cli/jobs/deepspeed/deepspeed-training/generate-yml.sh [hunk 3]
```diff
@@ -26,7 +32,7 @@ outputs:
 compute: azureml:gpu-v100-cluster
```
Comment (rtanase) [inline cli/jobs/deepspeed/deepspeed-training/generate-yml.sh:32]: how many GPUs per node does this cluster have? If it has 4, then we should initialize the variable holding the process count per node with 4..

File: cli/jobs/deepspeed/deepspeed-training/generate-yml.sh [hunk 4]
```diff
@@ -2,22 +2,28 @@
 # Generate key
 ssh-keygen -t rsa -f './src/generated-key' -N ''
 
-# Generate yaml file with key path
+# Pre-set process_count_per_instance so it can be passed into deepspeed via bash script.
```
Comment (rtanase) [inline cli/jobs/deepspeed/deepspeed-training/generate-yml.sh:5]: Should we also add that this number is supposed to represent the number of GPUs per VM based on the chosen cluster?

File: cli/jobs/deepspeed/deepspeed-autotuning/src/train.py [hunk 5]
```diff
@@ -130,10 +131,7 @@ def add_argument():
 #
 # Showcasing logging metrics to automl.
 if args.with_aml_log:
```
Comment (rtanase) [inline cli/jobs/deepspeed/deepspeed-autotuning/src/train.py:133]: should we remove these 2 lines? we're not logging to AutoML per se.. and not sure we actually need to log something.

File: cli/jobs/deepspeed/deepspeed-training/generate-yml.sh (hunk 3 above)
Comment (cassieesvelt) [reply cli/jobs/deepspeed/deepspeed-training/generate-yml.sh:32]: It has 8, should we use all 8 here?

File: cli/jobs/deepspeed/deepspeed-autotuning/src/train.py (hunk 5 above)
Comment (cassieesvelt) [reply cli/jobs/deepspeed/deepspeed-autotuning/src/train.py:133]: We don't need to log here, it was just to show that you can log metrics from the train.py file. But we also have the loss metric being logged so I can remove it. What do you mean by we're not logging to AutoML per se?

File: cli/jobs/deepspeed/deepspeed-autotuning/generate-yml.sh (hunk 2 above)
Comment (cassieesvelt) [reply cli/jobs/deepspeed/deepspeed-autotuning/generate-yml.sh]: yeah I think that makes it clearer, I renamed it

File: cli/jobs/deepspeed/deepspeed-autotuning/docker-context/Dockerfile (hunk 1 above)
Comment (cassieesvelt) [reply cli/jobs/deepspeed/deepspeed-autotuning/docker-context/Dockerfile]: I updated it, its just "pip install deepspeed -U" right?

File: cli/jobs/deepspeed/deepspeed-training/generate-yml.sh (hunk 4 above)
Comment (cassieesvelt) [reply cli/jobs/deepspeed/deepspeed-training/generate-yml.sh]: I changed the name, so its self-explanatory now

**Review by rtanase — APPROVED** [review]

**Review by savitamittal1 — APPROVED** [review]

**Pull request merged**